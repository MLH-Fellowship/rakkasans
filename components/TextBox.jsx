import React from 'react';
import { View, TextInput, StyleSheet } from 'react-native';

import Colors from '../constants/Colors';
import Icon from './Icon';
import Dimensions from '../constants/Dimensions';

export default function TextBox({ icon, ...otherProps }) {
  const styles = StyleSheet.create({
    container: {
      backgroundColor: Colors.lightGray,
      borderRadius: 5,
      flexDirection: 'row',
      justifyContent: 'space-between',
      padding: 15,
      marginVertical: 10,
      width: '100%',
    },
    icon: {
      marginLeft: 10,
    },
    textInput: {
      fontSize: 18,
    },
  });
  return (
    <View style={styles.container}>
      <TextInput style={[styles.textInput, Dimensions.font]} {...otherProps} />
      {icon && (
        <Icon style={styles.icon} name={icon} size={20} color={Colors.gray} />
      )}
    </View>
  );
}
