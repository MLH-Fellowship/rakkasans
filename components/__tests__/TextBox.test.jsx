import React from 'react';
import renderer from 'react-test-renderer';

import TextBox from '../TextBox';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<TextBox />).toJSON();
  expect(tree).toMatchSnapshot();
});
