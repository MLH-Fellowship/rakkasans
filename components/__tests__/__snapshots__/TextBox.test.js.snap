// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efefef",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "fontSize": 18,
        },
        Object {
          "fontFamily": "andale-mono",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
